### PR TITLE
Feature/dry datepickerinput

### PR DESCRIPTION
**Pull request by gcedo**
Extracted the `linkValue` behaviour in a mixin, small refactor here and there.

### CODE REVIEW

File: src/DatePickerInput.js [hunk 1]
```diff
@@ -85,46 +87,27 @@ const DatePickerInput = React.createClass({
...
     }
   },
 
   toggleDatePicker() {
-    if (this.state.showing) {
-      this.hide();
-    } else {
-      this.show();
-    }
+    let callback = this.state.showing ? this.props.onHide : this.props.onShow;
```
Comment (FrancescoCioria) [inline src/DatePickerInput.js]: `const` ? :tongue:

File: src/daypicker/DayPickerTop.js [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
 
-  changeMonth(month) {
-    this.props.visibleDate.month(month);
-    this.props.onChangeVisibleDate(this.props.visibleDate);
+  getInitialState() {
+    return { visibleDate: this.props.initialVisibleDate.clone() };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ visibleDate: nextProps.initialVisibleDate });
```
Comment (FrancescoCioria) [inline src/daypicker/DayPickerTop.js]: Why do we need internal state if `this.state.visibleDate` is always equal to `this.props.initialVisibleDate`?
Stateless components are easier to test, I suggest we use only `this.props.visibleDate` and delegate the state change to its father

File: src/DatePickerInput.js (hunk 1 above)
Comment (gcedo) [inline src/DatePickerInput.js]: Yep, my bad :)

File: src/daypicker/DayPickerTop.js (hunk 2 above)
Comment (gcedo) [inline src/daypicker/DayPickerTop.js]: Yup, I agree, I did not want the refactoring to be too much aggressive at once.

**Comment by FrancescoCioria** [comment]
Great job, thank you!

**Comment by giogonzo** [comment]
gotta love an on fire @gcedo :fire: :tractor:

**Comment by gcedo** [comment]
yep, and way better than a fired gcedo or fire on gcedo.

**Comment by FrancescoCioria** [comment]
closing in favor of #59

**Pull request closed without merging**